GPU/Common/TextureCacheCommon.cpp

```diff
@@ -539,75 +533,49 @@ void TextureCacheCommon::SetTexture(bool force) {
 	gstate_c.curTextureWidth = w;
 	gstate_c.curTextureHeight = h;
 
-	// Before we go reading the texture from memory, let's check for render-to-texture.
-	// We must do this early so we have the right w/h.
-	entry->framebuffer = nullptr;
-	if (Memory::IsDepthTexVRAMAddress(texaddr)) {
-		entry->status |= TexCacheEntry::STATUS_DEPTH;
-	}
-
-	AttachFramebufferToEntry(entry, 0);
-
-	// If we ended up with a framebuffer, attach it - no texture decoding needed.
-	if (entry->framebuffer) {
-		SetTextureFramebuffer(entry, entry->framebuffer);
-	}
-
 	nextTexture_ = entry;
-	nextNeedsRehash_ = entry->framebuffer == nullptr;
+	nextNeedsRehash_ = false;
 	// We still need to rebuild, to allocate a texture.  But we'll bail early.
 	nextNeedsRebuild_ = true;
+	return entry;
 }
 
-bool TextureCacheCommon::AttachFramebufferToEntry(TexCacheEntry *entry, u32 texAddrOffset) {
+std::vector<AttachCandidate> TextureCacheCommon::GetFramebufferCandidates(const TextureDefinition &entry, u32 texAddrOffset) {
```

Comment:
Should probably remove INEXACT from the logic now, since it should never happen?

-[Unknown]

Reply:
Actually, can remove more cases than that...